AppCenterDataStorage/AppCenterDataStorage/Internal/Util/MSDocumentUtils.m

```diff
@@ -139,23 +132,47 @@ + (MSDocumentWrapper *)documentWrapperFromDictionary:(NSObject *)object document
   NSDate *lastUpdatedDate = [NSDate dateWithTimeIntervalSince1970:[(NSNumber *)dictionary[kMSDocumentTimestampKey] doubleValue]];
   NSString *eTag = dictionary[kMSDocumentEtagKey];
   NSString *partition = dictionary[kMSPartitionKey];
-  return [self documentWrapperFromDictionary:object
+  return [self documentWrapperFromDictionary:dictionary
                                 documentType:documentType
                                         eTag:eTag
                              lastUpdatedDate:lastUpdatedDate
                                    partition:partition
                                   documentId:documentId
-                            pendingOperation:nil];
+                            pendingOperation:nil
+                             fromDeviceCache:fromDeviceCache];
 }
 
-+ (MSDocumentWrapper *)documentWrapperFromDictionary:(NSObject *)object
++ (BOOL)isSerializableDocument:(Class)classType {
+  return class_conformsToProtocol(classType, @protocol(MSSerializableDocument));
+}
+
++ (BOOL)isReferenceDictionaryWithKey:(nullable id)reference key:(NSString *)key keyType:(Class)keyType {
+
+  // Validate the reference is a dictionary.
+  if (!reference || ![(NSObject *)reference isKindOfClass:[NSDictionary class]]) {
```

Comment:
I believe you don't have to check `reference` for `nil` as there's no null reference error in Objective-C.